charts/portworx/templates/storage-cluster.yaml

```diff
@@ -0,0 +1,238 @@
+  {{- $managementInterface := .Values.managementInterface | default "none" }}
+  {{- $envVars := .Values.envVars | default "none" }}
+  {{- $customRegistryURL := .Values.customRegistryURL | default "none" }}
+  {{- $registrySecret := .Values.registrySecret | default "none" }}
+  {{- $licenseSecret := .Values.licenseSecret | default "none" }}
+  {{- $kvdbDevice := .Values.kvdbDevice | default "none" }}
+  {{- $miscArgs := include "px.miscArgs" . }}
+  {{- $volumesPresent := include "px.volumesPresent" . }}
+
+kind: StorageCluster
+apiVersion: core.libopenstorage.org/v1
+metadata:
+  name: "{{ required "Clustername cannot be empty" .Values.clusterName }}"
+  namespace: kube-system
+  annotations:
+    {{- if eq $openshiftInstall true }}
+    portworx.io/is-openshift: "true"
+    {{- end }}
+    {{- if eq $pksInstall true }}
+    portworx.io/is-pks: "true"
+    {{- end }}
+    {{- if eq $EKSInstall true }}
+    portworx.io/is-eks: "true"
+    {{- end }}
+    {{- if eq $AKSInstall true }}
+    portworx.io/is-aks: "true"
+    {{- end }}
+    {{- if (lookup "apps/v1" "Daemonset" "kube-system" "portworx") }}
+    portworx.io/migration-approved: "false"
+    {{- end }}
+    {{- if $miscArgs }}
+    portworx.io/misc-args: {{ $miscArgs | quote }}
+    {{- end }}
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+    app.kubernetes.io/instance: {{.Release.Name | quote }}
+spec:
+  image: portworx/oci-monitor:{{ required "A valid Image tag is required in the SemVer format" .Values.imageVersion }}
+  imagePullPolicy: Always
+  {{- if not (eq $customRegistryURL "none") }}
+  customImageRegistry: {{ $customRegistryURL }}
+  {{- end }}
+  {{- if not (eq $registrySecret "none") }}
+  imagePullSecret: {{ $registrySecret }}
+  {{- end }}
+
+  kvdb:
+  {{- if eq $internalKVDB true }}
+    internal: true
+  {{- else }}
+    internal: false
+    {{- if empty $etcdEndPoints }}
+    "{{ required "A valid ETCD url in the format etcd:http://<your-etcd-endpoint> is required. Verify that the key is correct and there isnt any typo in specifying that, also ensure it is accessible from all node of your kubernetes cluster" .etcdEndPoint}}"
+    {{- else }}
+    endpoints:
+      {{- $endpoints := $etcdEndPoints | split ";" }}
+      {{- range $key, $val := $endpoints }}
+      - {{$val}}
+      {{- end }}
+    {{- end }}
+    {{- if .Values.kvdb.authSecretName }}
+    authSecret: {{ .Values.kvdb.authSecretName }}
+    {{- end }}
+  {{- end }}
+
+{{- if and (ne $drives "none") (not (hasPrefix "/" $drives)) }}
+  cloudStorage:
+    deviceSpecs:
+      {{- $driveNames := $drives | split ";" }}
+      {{- range $index, $name := $driveNames }}
+      - {{ $name }}
+      {{- end }}
+{{- else }}
+  {{- if .AKSInstall }}
+  cloudStorage:
+    deviceSpecs:
+      - type=Premium_LRS,size=150
```

Comment:
Should we let user to customize the drive size?

Reply:
It wont be customizable here, for daemonset to operator migration we need to keep consistent.

for other customers with new installs, if we get new requests we can add this support by other values in values.yaml